akka-http-core/src/main/scala/akka/http/impl/engine/server/HttpServerBluePrint.scala

```diff
@@ -432,14 +435,14 @@ private[http] object HttpServerBluePrint {
         })
 
         setHandler(fromNet, new InHandler {
-          override def onPush(): Unit = sourceOut.push(grab(fromNet))
+          override def onPush(): Unit = sourceOut.push(grab(fromNet).bytes)
         })
         sourceOut.setHandler(new OutHandler {
           override def onPull(): Unit = {
             if (!hasBeenPulled(fromNet)) pull(fromNet)
             cancelTimeout(timeoutKey)
             sourceOut.setHandler(new OutHandler {
-              override def onPull(): Unit = pull(fromNet)
+              override def onPull(): Unit = if (!hasBeenPulled(fromNet)) pull(fromNet)
```

Comment:
For some reasons I saw tests failing here. Not sure why this PR would have changed the behavior.